src/Sql/Sql/help/Move-AzSqlInstanceDnsAlias.md

```diff
@@ -0,0 +1,432 @@
+---
+external help file: Microsoft.Azure.PowerShell.Cmdlets.Sql.dll-Help.xml
+Module Name: Az.Sql
+online version:
+schema: 2.0.0
+---
+
+# Move-AzSqlInstanceDnsAlias
+
+## SYNOPSIS
+Modifies the managed instance to which Azure SQL Managed Instance DNS Alias is pointing.
+
+## SYNTAX
+
+### MoveByNameAndSourceResourceIdParameterSet (Default)
+```
+Move-AzSqlInstanceDnsAlias [-ResourceGroupName] <String> [-InstanceName] <String> [-SourceResourceId] <String>
```

Comment:
It is the current resource ID of the DNS alias (old one). After execution of the Move command, that same alias will point to the target managed instance.

Reply:
So the ResourcegroupName and InstanceName here is used for target managed instance.

Reply:
That's right

Reply:
I'm wondering whether it would be more friendly to customer if we use parameter ResourceId to name current resource ID and use   DestinationResourceGroupName and DestinationInstanceName to name ResourcegroupName and InstanceName? We have some similar cmdlets with this pattern. See [Move-AzResource](https://docs.microsoft.com/en-us/powershell/module/az.resources/move-azresource?view=azps-8.1.0), [Move-AzDataLakeGen2Item](https://docs.microsoft.com/en-us/powershell/module/az.storage/move-azdatalakegen2item?view=azps-8.1.0), ...

Reply:
There is a problem here. All SQL cmdlets inherit from AzureSqlCmdletBase, and there exists a mandatory parameter called ResourceGroupName (that is why it is overridden here in the code). If I change base class, it will induce breaking changes in other cmdlets. But I could add the alias for the parameter. What do you think about adding the alias?

Reply:
I think you can hide the ResourceGroupName from base class by `private new string ResourceGroupName { get; set; }`, could you have a try?

Reply:
It cannot be hidden that way- it won't not compile because of https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs0114?f1url=%3FappId%3Droslyn%26k%3Dk(CS0114) which is treated as an error here.

It can be overridden by specifying public override string [Mandatory = false] ResourceGroupName, and then declaring a new mandatory parameter DestResourceGroupName.
However, I am not sure if this is a good practice, because we declare an unused field, just in order to hide it. Wouldn't declaring an alias for ResourceGroupName be descriptive enough?

Reply:
Will leave the alias then :)

Reply:
Hi, I tried again to hide the property and it worked. It seems I ommited the "new" keyword here. I will check how it looks now, and will update the PR. Thanks for the suggestion!

Reply:
Please let me know this PR is ready to review by adding label `needs-review`

Reply:
I've pushed new commits and added the label, please review :)